docs/design/README.md

```diff
@@ -108,8 +108,16 @@ and if this signal became unhandled current process will be terminated. \
 On POSIX compatible platforms keyboard handler will register it's own signal handler for the 
 `SIGINT` to be able to return terminal input in canonical mode after abnormal program termination 
 via pressing `Ctrl+C`. This signal handling could overlap with registered specific `SIGINT` 
-handler on upper level or inside keyboard handler's clients. It could cause side effects and 
-considering as current design and implementation limitations. 
+handler on upper level or inside keyboard handler's clients. To avoid this overlap keyboard handler
+will save previous `SIGINT` handler and call it after processing its own signal handler. It will 
+also restore the previous `SIGINT` handler in destructor. The keyboard handler constructor has an
+optional parameter `bool handle_sigint` which by default is set to `true`. If this parameter
```

Comment:
It's called `install_signal_handler` and not `handle_sigint`: https://github.com/ros-tooling/keyboard_handler/blob/f939a5c562408e32b65c485b51ffea35c30f3cf5/keyboard_handler/include/keyboard_handler/keyboard_handler_unix_impl.hpp#L55

Reply:
Good catch!. Fixed.